### PR TITLE
refactor: Simplify the search app and page logics

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/remote-debugger.js [hunk 1]
```diff
@@ -360,6 +305,61 @@ class RemoteDebugger extends events.EventEmitter {
     }
   }
 
+  async searchApp (currentUrl, maxTries, ignoreAboutBlankUrl) {
+    for (let i = 0; i < maxTries; i++) {
```
Comment (mykola-mokhnach) [inline lib/remote-debugger.js]: how about using retryInterval from asyncbox?

File: lib/remote-debugger.js [hunk 2]
```diff
@@ -360,6 +305,61 @@ class RemoteDebugger extends events.EventEmitter {
...
+      log.debug(`Trying out the possible app ids: ${possibleAppIds.join(', ')}`);
+      for (const attemptedAppIdKey of possibleAppIds) {
+        try {
+          log.debug(`Selecting app ${attemptedAppIdKey} (try #${i + 1} of ${maxTries})`);
+          const [appIdKey, pageDict] = await this.rpcClient.selectApp(attemptedAppIdKey, this.onAppConnect.bind(this));
+          // in iOS 8.2 the connect logic happens, but with an empty dictionary
+          // which leads to the remote debugger getting disconnected, and into a loop
+          if (_.isEmpty(pageDict)) {
+            log.debug('Empty page dictionary received. Trying again.');
+            continue;
```
Comment (mykola-mokhnach) [inline lib/remote-debugger.js]: should we also wait here?

File: lib/remote-debugger.js (hunk 1 above)
Comment (umutuzgur) [reply lib/remote-debugger.js]: I'm thinking about that. I need to introduce some error handling thou. Will do that afterwards

File: lib/remote-debugger.js (hunk 2 above)
Comment (umutuzgur) [reply lib/remote-debugger.js]: We can but it can wait for retryInterval implementation

**Comment by imurchie** [comment]
💯

**Pull request merged**